File: WebApi/Web.Api/Models/Recipe.cs
namespace ModerationKitchen.Web.Api.Models;

public record Recipe(
    string Slug,
    bool IsDraft,
    string Title,
    string Author,
    DateTime Date,
    string Intro,
    string HeroImage,
    string Body,
    RecipeRating Rating,
    string PrepTime,
    string CookTime,
    string QuantitySizeMade,
    IReadOnlyList<string> Ingredients,
    IReadOnlyList<string> Method,
    IReadOnlyList<string> Tags,
    IReadOnlyList<RecipeComment> Comments
);

File: src/app/core/interfaces/recipe.ts
import { RecipeComment } from "./recipe-comment";
import { RecipeRating } from "./recipe-rating";

export interface Recipe {
  slug: string;
  isDraft: boolean;
  title: string;
  author: string;
  date: Date;
  intro: string;
  heroImage: string;
  body: string;
  rating?: RecipeRating;
  prepTime: string;
  cookTime: string;
  quantitySizeMade: string;
  ingredients: string[];
  method: string[];
  tags: string[];
  comments: RecipeComment[];
}
